masonite/drivers/QueueAsyncDriver.py

```diff
@@ -19,22 +23,60 @@ def __init__(self, app: App):
         """
         self.container = app
 
-    def push(self, *objects, args=(), callback='handle', ran=1, channel=None):
+    def _threading(self):
+        """ Implements Async by Threading """
+
+        # Necessary for Python 3.4
+        if self.workers is None:
+            # Use this number because ThreadPoolExecutor is often
+            # used to overlap I/O instead of CPU work.
+            self.workers = (os.cpu_count() or 1) * 5
+        if self.workers <= 0:
+            raise QueueException("max_workers must be greater than 0")
+
+        with ThreadPoolExecutor(max_workers=self.workers) as executor:
+            for obj in self.objects:
+                if inspect.isclass(obj):
+                    obj = self.container.resolve(obj)
+
+                try:
+                    executor.submit(
+                        fn=getattr(obj, self.callback), args=self.args, kwargs=self.kwargs)
+                except AttributeError:
+                    # Could be wanting to call only a method asyncronously
+                    executor.submit(fn=obj, args=self.args, kwargs=self.kwargs)
+
+    def _multiprocessing(self):
+        """ Implements Async by Multiprocesses """
+
+        with ProcessPoolExecutor(max_workers=self.workers) as executor:
+            for obj in self.objects:
+                if inspect.isclass(obj):
+                    obj = self.container.resolve(obj)
+
+                try:
+                    executor.submit(
+                        fn=getattr(obj, self.callback), args=self.args, kwargs=self.kwargs)
+                except AttributeError:
+                    # Could be wanting to call only a method asyncronously
+                    executor.submit(fn=obj, args=self.args, kwargs=self.kwargs)
+
+    def push(self, *objects, args=(), kwargs={}, callback='handle', mode='threading', workers=None):
```

Comment:
This has been in the back of my head for a while .. but i think we should experiment with making these `**kwargs` here. we may need to change `kwargs` here to `keywords` and then make a `**kwargs` at the end. Does that make sense? Or something similiar.

Here is an example definition:

```
    def push(self, *objects, args=(), kwargs={}, **keyword_args):
        """Push objects onto the async stack.

        Arguments:
            objects {*args of objects} - This can be several objects as parameters into this method.
        """

        self.workers = keyword_args.get('workers', None)
        self.objects = objects
        self.callback = keyword_args.get('callback', 'handle') # get callback keyword or `handle` as default argument
        self.args = args
        self.kwargs = kwargs
```

Reply:
reason being that a driver may need something weird like `hop=3` or something wacky. If we have `**keyword_args` then all drivers will simply ignore that but the actual driver that needs it will be able to use it.